proxy/src/coco.rs

```diff
@@ -101,15 +77,58 @@ impl Peer {
         Ok(f(&api))
     }
 
-    /// Returns the list of [`project::Project`] known for the configured [`paths::Paths`].
+    /// Returns the list of [`project::Project`]s known for the configured [`paths::Paths`].
     ///
     /// # Errors
     ///
     /// The function will error if:
     ///   * A lock was poisioned. See [`Self::with_api`].
+    #[allow(
+        clippy::wildcard_enum_match_arm,
+        clippy::match_wildcard_for_single_variants
+    )]
     pub fn list_projects(&self) -> Result<Vec<project::Project<entity::Draft>>, error::Error> {
-        let projects = self.projects.lock().map_err(|_| error::Error::LibradLock)?;
-        Ok(projects.values().cloned().collect())
+        self.with_api(|api| {
+            let storage = api.storage();
+            Ok(storage
+                .entities()?
+                .into_iter()
+                .flat_map(|entity| {
+                    entity.try_map(|info| match info {
+                        entity::data::EntityInfo::Project(info) => Some(info),
+                        _ => None,
+                    })
+                })
+                .collect())
+        })
+        .flatten()
+    }
+
+    /// Returns the list of [`user::User`]s known for the configured [`paths::Paths`].
+    ///
+    /// # Errors
+    ///
+    /// The function will error if:
+    ///   * A lock was poisioned. See [`Self::with_api`].
+    #[allow(
+        clippy::wildcard_enum_match_arm,
+        clippy::match_wildcard_for_single_variants
+    )]
+    pub fn list_users(&self) -> Result<Vec<user::User<entity::Draft>>, error::Error> {
```

Comment:
Are we using this anywhere?

Reply:
Not yet, but I'm sure it'll be useful for feed-like features.

Reply:
Either we cover it with a test or we add when we need it, otherwise just feels like a liability.